rkyv/examples/remote_types.rs

```diff
@@ -0,0 +1,118 @@
+use rancor::Failure;
+use rkyv::{with::With, Archive, Deserialize, Serialize};
+
+// Assume this is a remote module or crate that you cannot modify.
+mod remote {
+    // Notably, this type does not implement the rkyv traits
+    #[derive(Debug, PartialEq)]
+    pub struct Foo {
+        pub ch: char,
+        pub bytes: [u8; 4],
+        pub _uninteresting: u32,
+        // ... and even has private fields
+        bar: Bar<i32>,
+    }
+
+    #[derive(Debug, PartialEq)]
+    pub struct Bar<T>(pub T);
+
+    impl Foo {
+        // A constructor which is necessary for deserialization because there
+        // are private fields.
+        pub fn new(
+            ch: char,
+            bytes: [u8; 4],
+            _uninteresting: u32,
+            bar: Bar<i32>,
+        ) -> Self {
+            Self {
+                ch,
+                bytes,
+                _uninteresting,
+                bar,
+            }
+        }
+
+        // The getter for a private field.
+        pub fn bar(&self) -> &Bar<i32> {
+            &self.bar
+        }
+    }
+}
+
+// Let's create a local type that will serve as `with`-wrapper for `Foo`.
+// Fields must have the same name and type but it's not required to define all
+// fields.
+#[derive(Archive, Serialize, Deserialize)]
+#[rkyv(remote = remote::Foo)] // <-
+#[rkyv(archived = ArchivedFoo)]
```

Comment:
I really like this idea; maybe the default name for archived types should be made from the remote type. I’ll make a quick change to add this.